src/mlpack/core/data/scaler_methods/zcawhitening.hpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file zcawhitening.hpp
+ * @author Jeffin Sam
+ *
+ * Whitening scaling to scale features, Using ZCA Whitening.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_ZCA_WHITENING_SCALE_HPP
+#define MLPACK_CORE_DATA_ZCA_WHITENING_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/lin_alg.hpp>
+#include <mlpack/core/data/scaler_methods/pcawhitening.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple ZcaWhitening class.
+ *
+ * Whitens a matrix using the eigendecomposition of the covariance matrix.
+ * Whitening means the covariance matrix of the result is the identity matrix.
+ * 
+ * For whitening related formula and more info, check the link below.
+ * http://ufldl.stanford.edu/tutorial/unsupervised/PCAWhitening/
+ *
+ * @code
+ * arma::mat input;
+ * Load("train.csv", input);
+ * arma::mat output;
+ *
+ * // Fit the features.
+ * ZcaWhitening scale;
+ * scale.Fit(input)
+ *
+ * // Scale the features.
+ * scale.Transform(input, output);
+ *
+ * // Retransform the input.
+ * scale.InverseTransform(output, input);
+ * @endcode
+ */
+class ZcaWhitening
+{
+ public:
+  /**
+  * A constructor to set the regulatization parameter.
+  *
+  * @param eps Regularization parameter.
+  */
+  ZcaWhitening(double eps = 0.00005)
+  {
+    epsilon = eps;
+  }
+  /**
+  * Function to fit features, to find out the min max and scale.
+  *
+  * @param input Dataset to fit.
+  */
+  template<typename MatType>
+  void Fit(const MatType& input)
+  {
+    data::PcaWhitening scale(epsilon);
+    scale.Fit(input);
+    // Important to store results, so that i can use in InverseTranform().
+    eigenVectors = scale.EigenVectors();
+    itemMean = scale.ItemMean();
+    eigenValues = scale.EigenValues();
```

Comment:
Actually you might even be able to simplify this more by just holding a `PcaWhitening` object in this class, so you don't have to extract the eigenvectors, item mean, and eigenvalues.  Up to you what you want to do though.

Reply:
That was a really good idea, Didn't think that way. Thanks for the idea.